base_view_inheritance_extension/models/ir_ui_view.py

```diff
@@ -0,0 +1,161 @@
+            '%s(%s)' % (
+                self,
+                ','.join(
+                    [
+                        UnquoteObject(
+                            a if not isinstance(a, basestring)
+                            else "'%s'" % a
+                        )
+                        for a in args
+                    ] +
+                    [
+                        '%s=%s' % (UnquoteObject(k), v)
+                        for (k, v) in kwargs.iteritems()
+                    ]
+                )
+            )
+        )
+
+
+class UnquoteEvalObjectContext(tools.misc.UnquoteEvalContext):
+    def __missing__(self, key):
+        return UnquoteObject(key)
+
+
+class IrUiView(models.Model):
+    _inherit = 'ir.ui.view'
+
+    @api.model
+    def apply_inheritance_specs(self, source, specs_tree, inherit_id):
+        for specs, handled_by in self._iter_inheritance_specs(specs_tree):
+            source = handled_by(source, specs, inherit_id)
+        return source
+
+    @api.model
+    def _iter_inheritance_specs(self, spec):
+        if spec.tag == 'data':
+            for child in spec:
+                for node, handler in self._iter_inheritance_specs(child):
+                    yield node, handler
+            return
+        if spec.get('position') == 'attributes':
+            for child in spec:
+                node = etree.Element(spec.tag, **spec.attrib)
+                node.insert(0, child)
+                yield node, self._get_inheritance_handler_attributes(
+                    child
+                )
+            return
+        yield spec, self._get_inheritance_handler(spec)
+
+    @api.model
+    def _get_inheritance_handler(self, node):
+        handler = super(IrUiView, self).apply_inheritance_specs
+        if hasattr(
+            self, 'inheritance_handler_%s' % node.tag
+        ):
+            handler = getattr(
+                self,
+                'inheritance_handler_%s' % node.tag
+            )
+        return handler
+
+    @api.model
+    def _get_inheritance_handler_attributes(self, node):
+        handler = super(IrUiView, self).apply_inheritance_specs
+        if hasattr(
+            self, 'inheritance_handler_attributes_%s' % node.get('operation')
+        ):
+            handler = getattr(
+                self,
+                'inheritance_handler_attributes_%s' % node.get('operation')
+            )
+        return handler
+
+    @api.model
+    def inheritance_handler_attributes_python_dict(
+        self, source, specs, inherit_id
+    ):
+        """Implement
+        <$node position="attributes">
+            <attribute name="$attribute" operation="python_dict" key="$key">
+                $keyvalue
+            </attribute>
+        </$node>"""
+        node = self.locate_node(source, specs)
+        for attribute_node in specs:
+            python_dict = tools.safe_eval(
+                node.get(attribute_node.get('name')) or '{}',
+                UnquoteEvalObjectContext()
```

Comment:
this is where travis' failure comes from, add parameter `nocopy=True`, that should fix the test run

Reply:
it gives a warning, not a failure in runbot full logs

@lfreeke, could you attend @hbrun comment?